src/test/java/org/opensearch/neuralsearch/query/HybridQueryIT.java

```diff
@@ -692,6 +698,101 @@ public void testWrappedQueryWithFilter_whenIndexAliasHasFilters_thenSuccess() {
         }
     }
 
+    @SneakyThrows
+    public void testConcurrentSearchWithMultipleSlices_whenSingleShardIndex_thenSuccessful() {
+        try {
+            updateClusterSettings(CONCURRENT_SEGMENT_SEARCH_ENABLED, true);
+            int numberOfDocumentsInIndex = 1_000;
+            initializeIndexIfNotExist(TEST_INDEX_DOC_QTY_ONE_SHARD, SINGLE_SHARD, numberOfDocumentsInIndex);
+            createSearchPipelineWithResultsPostProcessor(SEARCH_PIPELINE);
+
+            HybridQueryBuilder hybridQueryBuilder = new HybridQueryBuilder();
+            hybridQueryBuilder.add(QueryBuilders.matchAllQuery());
+
+            // first query with cache flag executed normally by reading documents from index
+            Map<String, Object> firstSearchResponseAsMap = search(
+                TEST_INDEX_DOC_QTY_ONE_SHARD,
+                hybridQueryBuilder,
+                null,
+                numberOfDocumentsInIndex,
+                Map.of("search_pipeline", SEARCH_PIPELINE)
+            );
+
+            int queryHitCount = getHitCount(firstSearchResponseAsMap);
+            assertEquals(numberOfDocumentsInIndex, queryHitCount);
+
+            List<Map<String, Object>> hitsNestedList = getNestedHits(firstSearchResponseAsMap);
+            List<String> ids = new ArrayList<>();
+            List<Double> scores = new ArrayList<>();
+            for (Map<String, Object> oneHit : hitsNestedList) {
+                ids.add((String) oneHit.get("_id"));
+                scores.add((Double) oneHit.get("_score"));
+            }
+
+            // verify that scores are in desc order
+            assertTrue(IntStream.range(0, scores.size() - 1).noneMatch(idx -> scores.get(idx) < scores.get(idx + 1)));
+            // verify that all ids are unique
+            assertEquals(Set.copyOf(ids).size(), ids.size());
+
+            Map<String, Object> total = getTotalHits(firstSearchResponseAsMap);
+            assertNotNull(total.get("value"));
+            assertEquals(numberOfDocumentsInIndex, total.get("value"));
+            assertNotNull(total.get("relation"));
+            assertEquals(RELATION_EQUAL_TO, total.get("relation"));
+        } finally {
+            wipeOfTestResources(TEST_INDEX_DOC_QTY_ONE_SHARD, null, null, SEARCH_PIPELINE);
+            updateClusterSettings(CONCURRENT_SEGMENT_SEARCH_ENABLED, false);
+        }
+    }
+
+    @SneakyThrows
+    public void testConcurrentSearchWithMultipleSlices_whenMultipleShardsIndex_thenSuccessful() {
+        try {
+            updateClusterSettings(CONCURRENT_SEGMENT_SEARCH_ENABLED, true);
+            int numberOfDocumentsInIndex = 2_000;
```

Comment:
nit: 2000

Reply:
To me syntax with delimiter is more readable